diff --git a/src/config/api-server/vnc_cfg_ifmap.py b/src/config/api-server/vnc_cfg_ifmap.py
index e5dba48..4f3d20f 100644
--- a/src/config/api-server/vnc_cfg_ifmap.py
+++ b/src/config/api-server/vnc_cfg_ifmap.py
@@ -791,7 +791,7 @@ class VncKombuClient(object):
     def _init_server_conn(self, rabbit_ip):
         while True:
             try:
-                self._conn = kombu.Connection('amqp://guest:guest@%s:5672//' %(rabbit_ip))
+                self._conn = kombu.Connection('amqp://guest:contrail123@127.0.0.1:5672//')
                 self._obj_update_q = self._conn.SimpleQueue(self._update_queue_obj)
 
                 old_subscribe_greenlet = self._dbe_oper_subscribe_greenlet

