--- /opt/stack/contrail/third_party/zookeeper-3.4.6/bin/zkServer.sh-orig	2014-02-10 12:36:12.138312109 +0100
+++ /opt/stack/contrail/third_party/zookeeper-3.4.6/bin/zkServer.sh	2014-02-10 12:35:07.717084287 +0100
@@ -97,6 +97,17 @@
 
 _ZOO_DAEMON_OUT="$ZOO_LOG_DIR/zookeeper.out"
 
+JVM_OPTS=""
+if [ "$MAX_HEAP_SIZE"]; then
+    JVM_OPTS="$JVM_OPTS -Xms${MAX_HEAP_SIZE}"
+fi
+if [ "$MAX_HEAP_SIZE" ]; then
+    JVM_OPTS="$JVM_OPTS -Xmx${MAX_HEAP_SIZE}"
+fi
+if [ "$HEAP_NEWSIZE" ]; then
+    JVM_OPTS="$JVM_OPTS -Xmn${HEAP_NEWSIZE}"
+fi
+
 case $1 in
 start)
     echo  -n "Starting zookeeper ... "
@@ -106,7 +117,7 @@
          exit 0
       fi
     fi
-    nohup $JAVA "-Dzookeeper.log.dir=${ZOO_LOG_DIR}" "-Dzookeeper.root.logger=${ZOO_LOG4J_PROP}" \
+    nohup $JAVA ${JVM_OPTS} "-Dzookeeper.log.dir=${ZOO_LOG_DIR}" "-Dzookeeper.root.logger=${ZOO_LOG4J_PROP}" \
     -cp "$CLASSPATH" $JVMFLAGS $ZOOMAIN "$ZOOCFG" > "$_ZOO_DAEMON_OUT" 2>&1 < /dev/null &
     if [ $? -eq 0 ]
     then
